Library/Homebrew/formula.rb

```diff
@@ -1126,6 +1126,10 @@ def link_overwrite?(path)
     end
   end
 
+  def shared_library_extension
+    ".dylib"
```

Comment:
Could method be empty here and the content moved to `Library/Homebrew/extend/os/mac/formula.rb`?

Reply:
Yes, I thought of that. But if you look at other places where we equivalent stuff, the main method is the Mac one, and then you override it for Linux. Because homebrew is mainly a Mac package manager.

Reply:
I like Dawid's suggestion, but we have the concept of a "generic" OS mode of operation, that is used to run `brew tests` (I forget under which circumstances exactly.) So there likely does need to be a generic implementation of `shared_library_extension`. It could be some nonce definition, like `.shared_library`, but it probably makes more sense to just leave it as `.dylib`.

Reply:
I'm fine with either the default being macOS or Linux here



> It could be some nonce definition, like `.shared_library`,

I'd rather we didn't do this, though.

Reply:
Let's leave the default as `.dylib` then.